### PR TITLE
Netrunner commands

### PR DESCRIPTION
**Pull request by distributive**
- Added Netrunner plugin and backend for searching the NetrunnerDB card database and creating custom cards
- Added Fuzzy plugin for fuzzy searching algorithms
- Added some additional functionality for handling embed colours

There's a certain amount left to be desired but this is functional and already does strictly more than Beanstalk.

### CODE REVIEW

File: src/Tablebot/Plugin/Netrunner/Custom.hs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    ("maxInf", fmap $ NrInt . (read :: String -> Int)),
+    ("keywords", fmap $ NrText . pack),
+    ("minSize", fmap $ NrInt . (read :: String -> Int)),
+    ("strength", fmap $ NrInt . (read :: String -> Int)),
+    ("text", fmap $ NrText . pack),
+    ("title", fmap $ NrText . pack),
+    ("trash", fmap $ NrInt . (read :: String -> Int)),
+    ("type", fmap $ NrText . pack),
+    ("unique", fmap $ NrBool . fuzzyBool)
+  ]
```
Comment (finnbar) [inline src/Tablebot/Plugin/Netrunner/Custom.hs:58]:
```suggestion
keys :: [(String, Maybe String -> Maybe NrData)]
keys =
  [ ("advancement", fmap $ NrInt . readInt),
    ("points", fmap $ NrInt . readInt),
    ("link", fmap $ NrInt . readInt),
    ("cost", fmap $ NrInt . readInt),
    ("faction", fmap $ NrText . pack),
    ("influence", fmap $ NrInt . readInt),
    ("flavour", fmap $ NrText . pack),
    ("maxInf", fmap $ NrInt . readInt),
    ("keywords", fmap $ NrText . pack),
    ("minSize", fmap $ NrInt . readInt),
    ("strength", fmap $ NrInt . readInt),
    ("text", fmap $ NrText . pack),
    ("title", fmap $ NrText . pack),
    ("trash", fmap $ NrInt . readInt),
    ("type", fmap $ NrText . pack),
    ("unique", fmap $ NrBool . fuzzyBool)
  ]
  where
    readInt :: String -> Int
    readInt = read
```

File: src/Tablebot/Plugin/Netrunner/Custom.hs [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          influence_limit = nrToInt =<< lookup "maxInf" params,
+          keywords = nrToText =<< lookup "keywords" params,
+          minimum_deck_size = nrToInt =<< lookup "minSize" params,
+          pack_code = Nothing,
+          position = Nothing,
+          quantity = Nothing,
+          Card.side_code = Nothing,
+          strength = nrToInt =<< lookup "strength" params,
+          stripped_text = nrToText =<< lookup "text" params, -- TODO - strip this
+          stripped_title = nrToText =<< lookup "title" params, -- TODO - ditto
```
Comment (finnbar) [inline src/Tablebot/Plugin/Netrunner/Custom.hs:100]: How urgent are these TODOs?

**Review by finnbar — CHANGES_REQUESTED** [review]
Generally very cool, with some minor points:
* Card view doesn't display MU costs.
* Custom card help formatting is wrong.
* Netrunner help could list `{{! }}` and `{{| }}` options

Otherwise, very impressive work!

File: src/Tablebot/Plugin/Netrunner/Custom.hs (hunk 2 above)
Comment (distributive) [reply src/Tablebot/Plugin/Netrunner/Custom.hs:103]: Not at all; the stripped field is required to interface with the nrdb api (unless I want to write my own JSON parser for it). All they do is take the text and title fields and remove newlines and such.

Comment (distributive) [reply src/Tablebot/Plugin/Netrunner/Custom.hs:103]: Practically it doesn't even matter. If you find this comment chain while trying to work out what those lines are for, forget about them.

**Pull request merged**